velox/exec/OutputBuffer.h

```diff
@@ -74,6 +74,33 @@ class ArbitraryBuffer {
   std::deque<std::shared_ptr<SerializedPage>> pages_;
 };
 
+/// The data transferred by the destination buffer has two phases:
+/// 1. Buffered: the data resides in the buffer after enqueued and before
+///              acked / deleted.
+/// 2. Sent: the data is removed from the buffer after it is acked or
+///          deleted.
+struct DestinationBufferStats {
+  void recordEnqueue(const SerializedPage& data);
+
+  void recordAcknowledge(const SerializedPage& data);
+
+  void recordDelete(const SerializedPage& data);
+
+  /// A destination buffer is finished after it is deleted by the
+  /// OutputBuffer, so this flag is set by the OutputBuffer.
+  bool finished{false};
+
+  /// Stats of data that is currently buffered.
```

Comment:
Number of buffered bytes / rows / pages.

Reply:
done.

Reply:
Actually your proposed annotation is just like repeating the variables' names, which is not encouraged in our former discussions. 

This PR makes me quite confused about the annotation standard or method in velox. It seems that just mimicking existing annotations in velox would not work (as I did in several iters in this PR) because many existing annotations in velox don't match this PR's standard.

Anyway, I adjust the annotations to what you propose. Thanks for your detailed feedbacks and suggestions.

Reply:
@HolyLow This is a good point. I suggested a change because "Stats of data" is too generic and doesn't provide much insight on top of what variable names already provide. I think you have a good point that, perhaps, new comment is similarly redundant. Maybe it be be removed completely. I would keep it though since it helps to summarize the code below in a format that's a bit easier to read than dense code below it. I don't have a strong opinion here though except that we shouldn't use generic "Stats of data".